src/engine/sync/synccontrol.cpp

```diff
@@ -129,7 +127,21 @@ void SyncControl::setSyncMode(SyncMode mode) {
     // requires. Bypass confirmation by using setAndConfirm.
     m_pSyncMode->setAndConfirm(mode);
     m_pSyncEnabled->setAndConfirm(mode != SYNC_NONE);
-    m_pSyncMasterEnabled->setAndConfirm(isMaster(mode));
+    switch (mode) {
+    case SYNC_INVALID:
+    case SYNC_NONE:
+    case SYNC_FOLLOWER:
+        m_pSyncMasterEnabled->setAndConfirm(0);
```

Comment:
Did you consider to add an enum for these values?

Reply:
ok, and added a conversion func